src/Artsy/Analytics/Schema/Interaction.ts

```diff
@@ -35,3 +35,22 @@ export interface Interaction {
    */
   destination_path?: string
 }
+
+export interface AuthInteraction {
+  /*
+  * The action taken that prompted user to signup or login.
+  */
+  intent?: string
+
+  /*
+  * Flow
+  */
+  flow?: string
+
+  /*
+  *  The type of action that triggered the modal (eg: click, timed)
+  */
+  trigger?: string
+}
+
+export type Interaction = BaseInteraction & AuthInteraction
```

Comment:
I don’t think this is right, as it effectively adds these 3 optional properties to all places where an `Interaction` is used. I think you should export both, have `AuthInteraction` extend `BaseInteraction`, and in `src/Artsy/Analytics/Schema/index.ts` add `AuthInteraction` to the `Trackables` union.

I’m also not a big fan of naming things “base”, I would prefer `BaseInteraction` were named after what it’s used for. (The same applies to `AuthInteraction` btw, which I _think_ may be an interaction that led to an authentication prompt?)

I also wonder if indeed `intent` should really be optional or if you only had to do that to make type-checking work, which would presumably fail on all places where `Interaction` was already being used without `intent`? Once you make `AuthInteraction` its own thing (that extends `BaseInteraction`), this problem would go away and allow you to have required properties.

Reply:
@l2succes and I actually discussed this approach. I can't remember why we decided to combined the types though. I do think it makes sense to export `AuthenticationInteraction` separately and add it to the `Trackables` union. Luc do you have any objection?

`intent` is used just about everywhere except for when the CTA Banner is dismissed. For that reason I think we would want to make it optional.